File: TestSollanFrontend/src/app/account/admin-view/subject-management-view/subject.service.ts
import { Injectable } from "@angular/core";
import { BackendService } from "../../../general/backend/backend.service";
import { ThrowStmt } from "@angular/compiler";
import { BackendMappings } from "../../../general/utils/backendMappings";
import { Subject } from "../../../students/subject";
import { RequestParams } from "../../../general/utils/requestParams";

@Injectable()
export class SubjectService{



    constructor(private backendService: BackendService){

    }


    getAll(){
        return this.backendService.get(BackendMappings.Subject.ALL);
    }



    save(subject: Subject){
        return this.backendService.post(BackendMappings.Subject.ADD, subject);
    }

    remove(subject: Subject){
        return this.backendService.post(BackendMappings.Subject.REMOVE, subject);
    }

    byTeacherId(teacherId: string){
        let params = {};
        params[RequestParams.TEACHER_ID] = teacherId;
        return this.backendService.get(BackendMappings.Subject.BY_TEACHER_ID, params);
    }

    byClassId(classId: string){
        let params = {};
        params[RequestParams.CLASS_ID] = classId;
        return this.backendService.get(BackendMappings.Subject.BY_CLASS_ID, params);
    }

}

File: TestSollanBackend/src/main/java/com/sollan/subjects/service/SubjectService.java
package com.sollan.subjects.service;

import java.util.List;

import com.sollan.subjects.model.Subject;

public interface SubjectService {
	
	List<Subject> getAll();
	Subject byId(Long id);
	List<Subject> byIds(List<Long> ids);
	Subject save(Subject s);
	void delete(Subject s);
	List<Subject> byTeacherId(Long teacherId);
	List<Subject> byClassId(Long classId);

}
